expression/builtin_time.go

```diff
@@ -315,16 +343,31 @@ func (b *builtinTimeDiffSig) eval(row []types.Datum) (d types.Datum, err error)
 	if args[0].IsNull() || args[1].IsNull() {
 		return
 	}
+
 	sc := b.ctx.GetSessionVars().StmtCtx
-	t1, err := convertDatumToTime(sc, args[0])
+	fsp := math.Max(float64(args[0].Frac()), float64(args[1].Frac()))
```

Comment:
add int(xxx) here.
so we do not need to convert the type in line 349 and line 353